src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCancelByIdControllerTest.java

```diff
@@ -115,7 +130,120 @@ void should_succeed_and_return_204_and_update_cft_task_state() throws Exception
 
         createRoleAssignment(roleAssignments, roleAssignmentRequest);
         RoleAssignmentResource accessControlResponse = new RoleAssignmentResource(roleAssignments);
-        insertDummyTaskInDb(taskId);
+        TaskRoleResource tribunalResource = new TaskRoleResource(
+            "tribunal-caseworker", true, false, false, false, true,
+            true, new String[]{}, 1, false, "LegalOperations"
+        );
+        insertDummyTaskInDb(taskId, tribunalResource);
+
+        when(roleAssignmentServiceApi.getRolesForUser(
+            any(), any(), any()
+        )).thenReturn(accessControlResponse);
+
+        when(accessControlService.getRoles(IDAM_AUTHORIZATION_TOKEN))
```

Comment:
when you mock accessControlService.getRoles method this will not go down to roleAssignmentServiceApi.getRolesForUser which you have mocked in previous line. Same in all the other tests here

Reply:
Removed line